notebooks/sequence_and_schema.ipynb

```diff
@@ -0,0 +1,172 @@
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Chat APIs\n",
+    "\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /chat/vanilla\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->>Core: ChatHistory.messages[]\n",
+    "    Core->>LLM API: ChatHistory.messages[]\n",
+    "\n",
+    "```\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```mermaid\n",
+    "---\n",
+    "title: Data sequence - POST /chat/rag\n",
+    "---\n",
+    "\n",
+    "sequenceDiagram\n",
+    "    Django->> Core: ChatHistory.messages[], File[].uuid\n",
+    "    Core->>Embedding API: ChatHistory.messages[]\n",
+    "    Embedding API->>Core: ChatHistory.messages[].embeddings\n",
+    "    Elastic->>Core: File[].Chunk[].embeddings\n",
+    "    Core->>LLM API: ChatHistory.messages[].embeddings + File[].Chunk[].embeddings\n",
+    "\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Schema\n",
+    "\n",
+    "```mermaid\n",
+    "---\n",
+    "title: Django schema\n",
+    "---\n",
+    "\n",
+    "erDiagram\n",
+    "    User }|--|{ UserGroup : \"UserGroup.users\"\n",
+    "    User {\n",
+    "        UUID uuid\n",
+    "        string name\n",
+    "    }\n",
+    "    UserGroup {\n",
+    "        UUID uuid\n",
+    "        string name\n",
+    "        UUID[] users\n",
+    "    }\n",
+    "    \n",
+    "    FileRecord }|--|| UserGroup: \"FileRecord.owner\"\n",
+    "    FileRecord {\n",
+    "        UUID uuid\n",
+    "        UUID owner\n",
+    "        string key \n",
+    "    }\n",
+    "\n",
+    "    UserGroup ||--|{ Message : \"Mesage.owner\"\n",
+    "    Message {\n",
+    "        UUID uuid\n",
+    "        UUID owner\n",
+    "        string text\n",
+    "        string role \n",
+    "    }\n",
+    "\n",
+    "    Message }|--|| ChatHistory: \"ChatHistory.messages\"\n",
+    "    UserGroup ||--|{ ChatHistory: \"ChatHistory.owner\"\n",
+    "\n",
+    "    ChatHistory {\n",
+    "        UUID uuid\n",
+    "        string name\n",
+    "        UUID owner\n",
+    "        UUID[] messages\n",
+    "        UUID[] files_received\n",
+    "        UUID[] files_cited\n",
+    "    }\n",
+    "\n",
+    "    ChatHistory ||--o{ FileRecord: \"ChatHistory.files_received\"\n",
+    "    ChatHistory ||--o{ FileRecord: \"ChatHistory.files_cited\"\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "```mermaid\n",
+    "---\n",
+    "title: Elastic schema\n",
+    "---\n",
+    "\n",
+    "erDiagram\n",
+    "    File ||--o{ Chunk : \"File.uuid\"\n",
+    "    File {\n",
+    "        UUID uuid\n",
+    "    }\n",
+    "    Chunk {\n",
+    "        UUID uuid\n",
+    "        UUID parent_file_uuid\n",
+    "        int index\n",
+    "        str text\n",
+    "        dict metadata\n",
```

Comment:
What's in this metadata?

Reply:
https://github.com/Unstructured-IO/unstructured/blob/4656b8cbe5f1f0c8468a7f11ab8bb39e07fd6c94/unstructured/documents/elements.py#L154

Reply:
Cool. Would we be directly referencing anything from this metadata in code anywhere?

Reply:
Page_number will be used for citation to load up the specific page of a URL. For non paged files (like txt), we're likely to add an offset attribute to chunks (e.g. chunk 22/33 of FileX starts at 3456543 char offset into FileX). That will help us highlight text.

Reply:
we are in sterreamlit, would be good to get this nailed down, if only the mandatory fields (can always have `extra=true` to allow people to crowbar in additional stuff